apps/next/src/pages/AccountManagement/components/ImportKeystoreFile/components/KeystoreFilePassword.tsx

```diff
@@ -15,25 +27,38 @@ export const KeystoreFilePassword = ({
   file,
   onCancel,
 }: KeystoreFilePasswordProps) => {
+  const { capture } = useAnalyticsContext();
+  const { replace } = useHistory();
   const { t } = useTranslation();
   const theme = useTheme();
 
-  const { importFile, isImporting } = useImportKeystoreFile();
-
-  const [filePassword, setFilePassword] = useState('');
   const [unlockError, setUnlockError] = useState<string>();
+  const [filePassword, setFilePassword] = useState('');
 
-  const errorHandler = useCallback(() => {
-    setUnlockError(t('Failed to import the keystore file.'));
-  }, [t]);
+  const importCallbacks: Callbacks = {
+    onFailure: () => {
+      capture('KeystoreFileImportFailure');
+      setUnlockError(t('Failed to import the keystore file.'));
+    },
+    onStarted: () => {
+      capture('KeystoreFileImportStarted');
+    },
+    onSuccess: () => {
+      capture('KeystoreFileImportSuccess');
+      toast.success(t('Successfully imported the keystore file.'));
+      replace('/account-management');
+    },
+  };
+
+  const { importFile, isImporting } = useImportKeystoreFile(importCallbacks);
```

Comment:
Let's wrap it in `useMemo`, otherwise the callbacks will be redefined on every state change, making any hooks and callbacks inside `useImportKeystoreFile` to get redefined too :)